lib/Runtime/Library/JavascriptBigInt.cpp

```diff
@@ -397,4 +414,183 @@ namespace Js
         return (leftBigInt->Compare(rightBigInt) == 0);
     }
 
+    // pbi1 += pbi2 assume pbi1 > pbi2
+    void JavascriptBigInt::AddAbsolute(JavascriptBigInt * pbi1, JavascriptBigInt * pbi2)
+    {
+        digit_t carryDigit = 0;
+        digit_t *pDigit1 = pbi1->m_digits;
+        digit_t *pDigit2 = pbi2->m_digits;
+        digit_t i = 0;
+
+        for (; i < pbi2->m_length; i++)
+        {
+            digit_t tempCarryDigit = 0;
+            if (pDigit2[i] > 0)
+            {
+                pDigit1[i] = JavascriptBigInt::AddDigit(pDigit1[i], pDigit2[i], &tempCarryDigit);
+            }
+            if (carryDigit > 0)
+            {
+                pDigit1[i] = JavascriptBigInt::AddDigit(pDigit1[i], carryDigit, &tempCarryDigit);
+            }
+            carryDigit = tempCarryDigit;
+        }
+
+        for (; i < pbi1->m_length; i++)
+        {
+            digit_t tempCarryDigit = 0;
+            if (carryDigit > 0)
+            {
+                pDigit1[i] = JavascriptBigInt::AddDigit(pDigit1[i], carryDigit, &tempCarryDigit);
+            }
+            carryDigit = tempCarryDigit;
+        }
+
+        if (0 < carryDigit) // length increase by 1
+        {
+            if (pbi1->m_length >= pbi1->m_maxLength && !pbi1->Resize(pbi1->m_length + 1))
+            {
+                return;
+            }
+            pbi1->m_digits[pbi1->m_length++] = carryDigit;
+        }
+    }
+
+    // pbi1 -= pbi2 assume pbi1 > pbi2
+    void JavascriptBigInt::SubAbsolute(JavascriptBigInt * pbi1, JavascriptBigInt * pbi2)
+    {
+        digit_t carryDigit = 0;
+        digit_t *pDigit1 = pbi1->m_digits;
+        digit_t *pDigit2 = pbi2->m_digits;
+        digit_t i = 0;
+
+        for (; i < pbi2->m_length; i++)
+        {
+            digit_t tempCarryDigit = 0;
+            if (pDigit2[i] > 0)
+            {
+                pDigit1[i] = JavascriptBigInt::SubDigit(pDigit1[i], pDigit2[i], &tempCarryDigit);
+            }
+            if (carryDigit > 0)
+            {
+                pDigit1[i] = JavascriptBigInt::SubDigit(pDigit1[i], carryDigit, &tempCarryDigit);
+            }
+            carryDigit = tempCarryDigit;
+        }
+
+        for (; i < pbi1->m_length; i++)
+        {
+            digit_t tempCarryDigit = 0;
+            if (carryDigit > 0)
+            {
+                pDigit1[i] = JavascriptBigInt::SubDigit(pDigit1[i], carryDigit, &tempCarryDigit);
+            }
+            carryDigit = tempCarryDigit;
+            if (carryDigit == 0)
```

Comment:
This could go in the test clause of the `for` loop, to keep the control flow simpler. And AddAbsolute could use a similar check.

Reply:
will do